src/controller/middleware/renderMyPosts.js

```diff
@@ -1,10 +1,17 @@
 const getMyPosts = require('../../database/queries/getMyPosts');
 
 exports.renderMyPosts = (req, res, next) => {
-  getMyPosts(req.token.userId)
-    .then((result) => {
-      req.result = result.rows;
-      next();
-    })
-    .catch(e => next(e));
+  if (req.token) {
+    getMyPosts(req.token.userId)
+      .then((result) => {
+
+        req.result = result.rows;
+        next();
+      })
+      .catch((e) => {
+        next(e);
+      });
+  } else {
+    next();
```

Comment:
@naremanhilles I wonder if this `else` should be an error (i.e. return a 401)

Reply:
You are right :+1:   else {
    res.redirect('/');
  } now code correct.

Reply:
Cool, thanks for doing that!